shell/platform/linux/public/flutter_linux/fl_plugin.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_PLUGIN_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_PLUGIN_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <glib-object.h>
+
+G_BEGIN_DECLS
+
+G_DECLARE_DERIVABLE_TYPE(FlPlugin, fl_plugin, FL, PLUGIN, GObject)
+
+/**
+ * FlPlugin:
+ *
+ * #FlPlugin is FIXME. Plugins are added to a #FlPluginRegistry.
```

Comment:
From an API standpoint, they are registered with FlPluginRegistar, not added to the registry.

As for what it's for, it primarily exists to provide a bunch of hooks that a plugin can register to have called. See the iOS `FlutterPlugin`, and by extension `FlutterApplicationLifeCycleDelegate`, to see the kinds of things that we'll eventually want to add here.

That said, we don't necessarily need this class in the GTK embedding, and we should decide based on what's idiomatic for the platform. For ObjC, having an interface protocol that classes implement, and register using, is a very common way to register for callbacks and delegation. (Or at least, was; increasingly blocks are used instead.) What's the standard GLib pattern for registering as a delegate or to receive callbacks for specific events?

Reply:
I think using GObject interfaces is probably going to be the solution here, so each plugin will be a GObject that implements any interfaces that you want to subscribe to. GTK would normally use signals for things like this, but that's probably because they're easier to use than interfaces. Using interfaces will make it easier to keep adding new functionality over time.